toolchain/sem_ir/formatter.cpp

```diff
@@ -795,6 +796,15 @@ class Formatter {
       FormatType(fn.return_type_id);
     }
 
+    // TODO: Move this conversion of kind to string elsewhere.
```

Comment:
What about using EnumBase for BuiltinFunctionKind? Note, that'd also give you a name automatically if you were fine with `IntAdd` instead of `int.add`

Note, fine if it's not part of this PR

Reply:
Noted.

Part of the refactoring that I'm planning is to use a `.def` file to define the builtins, or something like that. I'm not yet sure whether we want `EnumBase` or something more like `IdBase` here -- Clang has a mechanism for dynamically adding builtins, which would suggest something more like `IdBase`, but we might not want to support that.